webpack/assets/javascripts/react_app/common/hooks/API/APIHooks.js

```diff
@@ -18,6 +18,7 @@ import { APIActions } from '../../../redux/API';
 export const useAPI = (method, url, options) => {
   const dispatch = useDispatch();
   const keyRef = useRef(options?.key);
+  const optionsRef = useRef(options);
```

Comment:
this is to solve the infinite loop of `useEffect` when options are passed, it took me a while to figure out why the request is stuck on `PENDING` when `options` with custom key are passed
I also don't see `useAPI` to be used with `options` anywhere in the existing codebase

Reply:
`useEffect` compares the values in the dependency array by reference (shallow), those causes the infinite loop.
in your case, `options` object is  constant, so just define it in the `constants.js` file:
```js
export const API_OPTIONS = { key: 'some-key'}
```
this will stop your infinite loop.

if options object is dynamic and based on other props, wrapping it with `useMemo` works:
```js
const options = useMemo( optionCreator, [prop]);
```